File: administracion-electoral-frontend/src/models/cargo.ts
import { BaseModel } from './base';
import { Seccion } from './seccion';

// Cargo types
export interface Cargo extends BaseModel {
  nombre: string;
  secciones_afectadas: Seccion[];
}

export interface CargoFormData {
  nombre: string;
  secciones_afectadas_ids: number[];
} 

File: administracion_electoral_backend/administracion_electoral/models/cargo.py
from django.db import models
from .seccion import Seccion

class Cargo(models.Model):
    nombre = models.CharField(max_length=100, unique=True)
    secciones_afectadas = models.ManyToManyField(
        Seccion, 
        related_name='cargos',
        help_text="Secciones donde este cargo está en disputa"
    )
    
    def __str__(self):
        return self.nombre
    
    class Meta:
        verbose_name = "Cargo"
        verbose_name_plural = "Cargos"
        ordering = ['nombre'] 